week3/index.js

```diff
@@ -1,15 +1,119 @@
-const Express = require('express')
-const bodyParser = require('body-parser')
+"use strict";
 
-const app = Express()
+const fs = require("fs");
+const express = require("express");
+const bodyParser = require("body-parser");
 
-app.use(bodyParser.json())
+const app = express();
+const todosFile = "todos.json";
+const PORT = 4000;
 
-const {list, create, update, remove} = require('./actions')
+app.listen(PORT, () => {
+    console.log(`Server listening on port ${PORT}`);
+});
+
+
+
+function handleRequest() {
+
+    app.use(bodyParser.json());
+
+
+    app.get("/list", (req, res) => {
+        readTodosFile().then(data => {
+            if (!data || data === "[]") res.send("There are no items on the list");
+  
+            res.send(data);
+
+        }).catch(err => console.error(err));   
```

Comment:
Nice job using a `.catch()` to handle promise rejections. One problem - since you only `res.send()` in the successful case, if you encounter an error that triggers the `catch()` and not the `then()`, the response is never sent (and the browser or client will just wait for the request forever, eventually timing out).

One pattern you could use is that in your error cases, you can send back a JSON document like:
```
{ "success": false, "error": "${ err }" }
```

Or something similar - this way your caller gets the information about the error too.

(If you want to get really fancy - you would also want to send back the appropriate response code: 4xx (meaning 400-something) for client errors (malformed request, request for an element that doesn't exist, etc) and 5xx for server errors like a code bug)